controllers/extensions/addon_controller_stages.go

```diff
@@ -1082,3 +1083,12 @@ func findDataKey[V string | []byte](data map[string]V, refObj extensionsv1alpha1
 	}
 	return false
 }
+
+func setAddonProviderAndVersion(ctx context.Context, stageCtx *stageCtx, addon *extensionsv1alpha1.Addon) {
```

Comment:
There should be a reverse logic that also needs to be supported, setting the corresponding label based on the version and provider fields in the spec if the specified labels do not exist.